test/src/org/labkey/test/tests/idri/FormulationsTest.java

```diff
@@ -312,30 +316,21 @@ protected void defineParticleSizeAssay()
 
         log("Defining Particle Size Assay");
         clickAndWait(Locator.linkWithText("Manage Assays"));
-        clickButton("New Assay Design");
+        ReactAssayDesignerPage assayDesignerPage = _assayHelper.createAssayDesign("Particle Size", PS_ASSAY)
+            .setDescription(PS_ASSAY_DESC);
 
-        assertTextPresent("Particle Size Data");
-        checkCheckbox(Locator.radioButtonByNameAndValue("providerName", "Particle Size"));
-        clickButton("Next");
+        assayDesignerPage.goToBatchFields();
+        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
```

Comment:
Avoid putting Locators in tests. This Locator should live in `DomainFormPanel`, if we need it at all. Only a domain designer specific test should really care that there is a special element when there are no fields.

The empty list returned by `DomainFormPanel.fieldNames()` provides the equivalent information for most scenarios and requires the test to know fewer domain designer implementation details.

Reply:
Good point. Updated in new fb.